pkg/cloudprovider/cloud.go

```diff
@@ -47,6 +48,8 @@ type Interface interface {
 	ScrubDNS(nameservers, searches []string) (nsOut, srchOut []string)
 	// HasClusterID returns true if a ClusterID is required and set
 	HasClusterID() bool
+	// SetInformers sets the informer on the cloud object. Also returns true if the interface is supported, false otherwise.
+	SetInformers(informerFactory informers.SharedInformerFactory) bool
```

Comment:
I would avoid making all cloud providers implement this. Either an optional interface gce can implement or a param to Initialize would make more sense

Reply:
I think we looked into this and due to the nature of where the informers must be set the SetInformers method must be generic to all cloud providers.

See CreateControllerContext in controllermanager.go:
A possible alternative is to only set informers on the cloud if "s.CloudProvider == 'gce'" by then doing a cast of the generic cloud object. I believe that it would be much more brittle this way and would prefer to keep it generic.

Reply:
> due to the nature of where the informers must be set the SetInformers method must be generic to all cloud providers.

That's a good use case for optional interfaces. Unrelated setters with empty implementations in all but one cloud provider is a code smell

Reply:
You can add a new interface, maybe something like "InformerUser" interface.  Then you can check if the cloud object implements it, and then call SetInformers.  You can see the other examples of interfaces in this file.

Reply:
Understood, I was unaware of this way of doing things. Thanks for the tip. I'll add it in now